src/sys/socket/addr.rs

```diff
@@ -1014,4 +1101,227 @@ pub mod sys_control {
             fmt::Display::fmt(self, f)
         }
     }
-}
\ No newline at end of file
+}
+
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[derive(Clone, Copy)]
+pub struct EtherAddr(pub libc::sockaddr_ll);
+
+#[cfg(any(target_os = "dragonfly",
+          target_os = "freebsd",
+          target_os = "ios",
+          target_os = "macos",
+          target_os = "netbsd",
+          target_os = "openbsd"))]
+#[derive(Clone, Copy)]
+pub struct EtherAddr(pub libc::sockaddr_dl);
+
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+impl EtherAddr {
+    pub fn new(sll: libc::sockaddr_ll) -> Option<EtherAddr> {
```

Comment:
Why do you even implement `new()`? None of the other types implement it, so I suggest you drop it. You have this implemented to return an `Option` and then just unwrap the option ever time anyways, so I think this function is useless.